doc/user/content/sql/create-sink/kafka.md

```diff
@@ -430,6 +432,62 @@ message delivery, you should ensure that:
 
 For more details, see [the Kafka documentation](https://kafka.apache.org/documentation/).
 
+### Partitioning
+
+{{< private-preview />}}
+
+By default, Materialize assigns a partition to each message using the following
+strategy:
+
+  1. Encode the message's key in the specified format.
+  2. If the format uses a Confluent Schema Registry, strip out the
+     schema ID from the encoded bytes.
+  3. Hash the remaining encoded bytes using [SeaHash].
+  4. Divide the hash value by the topic's partition count and assign the
+     remainder as the message's partition.
+
+If a message has no key, all messages are sent to partition 0.
+
+You can configure a custom partitioning strategy with the `PARTITION BY` option,
+which allows you to specify a SQL expression that computes a hash for each
+message that determines what partition to assign to the message, like so:
```

Comment:
Adopted the first part of this. I left the last bit in place, because I think "which determines what partition to assign to the message" reads more clearly (active vs passive) than "which determines the partition a message is assigned to."